proxy/src/main.rs

```diff
@@ -126,6 +141,19 @@ async fn main() -> anyhow::Result<()> {
         )));
     }
 
+    if let Some(metric_collection_endpoint) = &config.metric_collection_endpoint {
+        let hostname = hostname::get().unwrap().into_string().unwrap();
+
+        tasks.push(tokio::spawn(
+            proxy_metrics::collect_metrics(
+                metric_collection_endpoint,
+                config.metric_collection_interval,
```

Comment:
I think we should extract this into `Option<MetricConfig>`. interval is useless without endpoint and vice versa.